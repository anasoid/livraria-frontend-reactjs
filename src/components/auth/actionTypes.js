export const AUTHENTICATE = 'AUTHENTICATE';
export const AUTH_ERROR = 'AUTH_ERROR';
export const SIGNOUT = 'SIGNOUT';