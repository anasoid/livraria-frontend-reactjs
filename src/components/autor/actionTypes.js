export const GET_AUTORES = 'GET_AUTORES';
export const ADD_AUTOR = 'ADD_AUTOR';
export const REMOVE_AUTOR = 'REMOVE_AUTOR';