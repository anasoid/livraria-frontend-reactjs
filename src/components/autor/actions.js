import Helper from './helper';
import {GET_AUTORES} from './types';

export const getAutoresList = () => {
  const helper = new Helper();
  return dispatch => {
    return helper.getAutoresList()
      .then((data) => {
        return dispatch({type: GET_AUTORES, payload: data});
      }).catch();
  };
};

