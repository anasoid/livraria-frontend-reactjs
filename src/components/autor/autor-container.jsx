import React from 'react';
import AutorTable from './autor-table';

const AutorContainer = (props) => {
  return(
    <div>
      <h1>Novo Autor</h1>
      <form>
        <label htmlFor={'autorNome'}>Nome: </label>
        <input id={'autorNome'} name={'nome'}/>
        <br/>
        <label htmlFor={'autorEmail'}>Email: </label>
        <input id={'autorEmail'} name={'email'}/>
        <br/>
        <button id={'btnGravarAutor'} type={'submit'}>Gravar Autor</button>
      </form>
      <br/>
      <AutorTable/>
    </div>
  );
};

export default AutorContainer;