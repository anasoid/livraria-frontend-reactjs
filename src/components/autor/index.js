import React from 'react';
import AutorContainer from './autor-container';

const Autor = () => {
  return (
    <AutorContainer/>
  );
};

export default Autor;