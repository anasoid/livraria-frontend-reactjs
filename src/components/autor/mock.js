export const autores = {
  "autores": [
    {'nome': 'fulano', 'email': 'fulano@fulano.com'},
    {'nome': 'ciclano', 'email': 'ciclano@ciclano.com'}
  ]
};