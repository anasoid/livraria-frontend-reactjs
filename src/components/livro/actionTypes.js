export const GET_LIVROS = 'GET_LIVROS';
export const ADD_LIVRO = 'ADD_LIVRO';
export const REMOVE_LIVRO = 'REMOVE_LIVRO';
export const UPDATE_LIVRO = 'UPDATE_LIVRO';
export const SELECT_LIVRO = 'SELECT_LIVRO';