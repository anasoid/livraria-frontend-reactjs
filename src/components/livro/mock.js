export const livros = {
  livros: [
    {titulo: 'reactjs', isbn: '12345678', preco: 20.0, dtLancamento: '11/05/2018', autores: [{nome: 'fulano', email: 'fulano@email.com'}]},
    {titulo: 'react native', isbn: '87654321', preco: 15.0, dtLancamento: '20/08/2018', autores: [{nome: 'ciclano', email: 'ciclano@email.com'}]}
  ]
}