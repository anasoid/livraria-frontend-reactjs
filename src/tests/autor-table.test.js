import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import AutorTable from '../components/autor/autor-table';
import * as types from '../components/autor/actionTypes';


var Helper = require('../components/autor/helper').default;
Helper.prototype.getAutoresList = jest.fn(() => Promise.resolve(autores));

import initialState from './initial-state';
import {autores} from '../components/autor/mock';
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const expectedActions = [
  { type: types.GET_AUTORES }
];
const store = mockStore(initialState, expectedActions);

describe('test AutorTable components', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <AutorTable />
      </Provider>
    );
  });

  it('should renders without crashing', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <AutorTable />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('should renders without crashing with call componentWillMount', () => {

    // const mockFetch = jest.fn();
    //
    // const wrapper = mount(
    //   <Provider store={store} getAutoresList={mockFetch} >
    //     <AutorTable />
    //   </Provider>
    // );
    //
    // expect(wrapper).toBeDefined();
    // expect(mockFetch).toHaveBeenCalled();
  // });

  it('should be contain a table with title "Autores"', () => {
    const table = wrapper.find('table').at(0);
    expect(table.find('caption')).toHaveText('Autores');
  });

  it('should be contain a table with columns Nome, Email', () => {
    const rowsTh = wrapper.find('table').at(0).find('thead').find('tr').find('th');
    expect(rowsTh.at(0)).toHaveText('Nome');
    expect(rowsTh.at(1)).toHaveText('Email');
  });

});
