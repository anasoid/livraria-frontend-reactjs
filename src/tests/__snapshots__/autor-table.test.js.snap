// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test AutorTable components should renders without crashing 1`] = `
<div>
  <table>
    <caption>
      Autores
    </caption>
    <thead>
      <tr>
        <th>
          Nome
        </th>
        <th>
          Email
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
